Userland/Libraries/LibGUI/TextDocument.cpp

```diff
@@ -884,6 +884,40 @@ void RemoveTextCommand::undo()
     m_document.set_all_cursors(new_cursor);
 }
 
+ReplaceAllTextCommand::ReplaceAllTextCommand(GUI::TextDocument& document, String const& text, GUI::TextRange const& range)
+    : TextDocumentUndoCommand(document)
+    , m_text(text)
+    , m_range(range)
+{
+}
+
+void ReplaceAllTextCommand::redo()
+{
+    m_document.remove(m_range);
+    m_document.set_all_cursors(m_range.start());
+    auto new_cursor = m_document.insert_at(m_range.start(), m_text, m_client);
+    m_range.set_end(new_cursor);
+    m_document.set_all_cursors(new_cursor);
+}
+
+void ReplaceAllTextCommand::undo()
+{
+    m_document.remove(m_range);
+    m_document.set_all_cursors(m_range.start());
+    auto new_cursor = m_document.insert_at(m_range.start(), m_text);
+    m_document.set_all_cursors(new_cursor);
+}
+
+bool ReplaceAllTextCommand::merge_with(GUI::Command const&)
+{
+    return false;
+}
+
+String ReplaceAllTextCommand::action_text() const
+{
+    return "Playground format text";
```

Comment:
This text is specific to the Playground application, but we're in LibGUI here so we'll want to make this more generic.

Even better: we can make it customizable! The `ReplaceAllTextCommand` can take a constructor parameter with an override text. That way, we can set a custom undo item string whenever generating this command.